docs/articles/commands/custom_context_checks.md

```diff
@@ -59,7 +62,8 @@ public class DirectMessageUsageCheck : IContextCheck<DirectMessageUsageAttribute
 }
 ```
 
-As seen here, we return `null` when the command is allowed to be executed. If the command is not allowed to be executed, we return an error string which can be retrieved from the `CommandsExtension.CommandErrored` event.
+> [!WARNING]
+> Your check may inspect the command context to get more information, but it should not make any API calls, especially none that may alter state such as `RespondAsync`. This is an easy source of bugs, and depending on what you call you may stall an interaction over the three-second limit, which will break your interactions.
```

Comment:
This warning is incorrect. The command context passed to checks is the same context that's passed to the command. While it may be discouraged to have long-running checks, I would not phrase it as to make it sound like it'll cause random commands to break. The user can alter the state or do whatever they wish, that will not cause the commands/extension to break because of our code.

To be more specific, the user can expect any commands annotated with `RequiresDatabaseAuthenticationCheck` to have already deferred the response, thus having the command implementation to be based around that expected alternated state.

Reply:
problems arise when two or more checks can do that. this is a similar situation to event handlers, where we strongly discourage these patterns even though they're technically legal. i'm not opposed to rewording the warning, but this is also something we should advise users to be very cautious of